### PR TITLE
Add Get-GcProject cmdlet

### PR DESCRIPTION
**Pull request by quoctruong**
Also split the SQL tests since it's taking too long (AppVeyor limit is an hour).

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/472)

### CODE REVIEW

File: Google.PowerShell.IntegrationTests/CloudResourceManager/CloudResourceManager.GcProject.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿. $PSScriptRoot\..\GcloudCmdlets.ps1
+Install-GcloudCmdlets
+
+$script:powerShellProjectId = "gcloud-powershell-testing"
+$script:powerShellProjectTwoId = "gcloud-powershell-testing-2"
+$script:powerShellProjectName = "Google Powershell Testing"
+$script:powerShellProjectTwoName = "GCloud PowerShell Testing 2"
+
+Describe "Get-GcIamPolicyBinding" {
```
Comment (ILMTitan) [inline Google.PowerShell.IntegrationTests/CloudResourceManager/CloudResourceManager.GcProject.Tests.ps1]: Describe "Get-GcProject"/"Get-GcpProject" ?

File: Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    /// <para
+    ///     type="link"
+    ///     uri="(https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy#cloud_platform_projects)">
+    /// [Projects]
+    /// </para>
+    /// <para type="link" uri="(https://cloud.google.com/resource-manager/docs/using-labels)">
+    /// [Labels]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GcProject")]
```
Comment (ILMTitan) [inline Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs]: Shouldn't this be named GcpProject? GCP for Google Cloud Platform. I don't know what GC stands for, other than Garbage Collection.

File: Google.PowerShell.IntegrationTests/CloudResourceManager/CloudResourceManager.GcProject.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    It "should work" {
+        $projects = Get-GcProject
+        $project = $projects | Where-Object {$_.ProjectId -eq $powerShellProjectId}
```
Comment (ILMTitan) [inline Google.PowerShell.IntegrationTests/CloudResourceManager/CloudResourceManager.GcProject.Tests.ps1]: You could write this as 
`$project = $projects | Where ProjectId -eq $powerShellProjectId`
avoiding the curly braces and `$_.` feels cleaner.

File: Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public Hashtable Label { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Filter is case insensitive.
+            string filter = string.Empty;
```
Comment (ILMTitan) [inline Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs]: Split this into a couple of functions? One to get the filter, one make the request and return an enum of `Project`s? End with a call to `WriteObject(projectsEnum, true);`?

File: Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    ///   <code>PS C:\> Get-GcProject</code>
+    ///   <para>This command gets all the available Google Cloud projects.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject -Name "My Project"</code>
+    ///   <para>This command gets the project that has the name "My Project".</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject -Id "my-id"</code>
+    ///   <para>This command gets the project that has the Id "My Project".</para>
```
Comment (ILMTitan) [inline Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs]: *has the Id "my-id".

File: Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject -Label {"environment" = "test"}</code>
```
Comment (ILMTitan) [inline Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs]: Don't Hashtables need to start with `@{` ? Isn't what you have here a scriptBlock?

**Comment by quoctruong** [comment]
@ILMTitan PTAL

File: Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs [hunk 7]
```diff
@@ -39,7 +39,7 @@ namespace Google.PowerShell.CloudResourceManager
     /// [Labels]
     /// </para>
     /// </summary>
-    [Cmdlet(VerbsCommon.Get, "GcProject")]
+    [Cmdlet(VerbsCommon.Get, "GcpProject")]
     public class GetGcProjectCmdlet : CloudResourceManagerCmdlet
```
Comment (ILMTitan) [inline Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs]: Do we want the class name and the cmdlet name to be out of sync like this?

File: Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs [hunk 8]
```diff
@@ -74,44 +74,59 @@ public class GetGcProjectCmdlet : CloudResourceManagerCmdlet
...
+            WriteObject(GetProjects(filter));
+        }
+
+        /// <summary>
+        /// Constructs filter for the list request based on name, projectid and label.
+        /// </summary>
+        private string ConstructFilter(string name, string projectId, Hashtable labels)
+        {
             // Filter is case insensitive.
             string filter = string.Empty;
```
Comment (ILMTitan) [inline Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs]: StringBuilder?

**Comment by quoctruong** [comment]
@ILMTitan PTAL

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**